src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/WorkTypesController.java

```diff
@@ -0,0 +1,102 @@
+import uk.gov.hmcts.reform.wataskmanagementapi.domain.entities.task.WorkType;
+import uk.gov.hmcts.reform.wataskmanagementapi.services.TaskManagementService;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import static org.slf4j.LoggerFactory.getLogger;
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+import static uk.gov.hmcts.reform.wataskmanagementapi.config.SecurityConfiguration.AUTHORIZATION;
+
+@SuppressWarnings({"PMD.DataflowAnomalyAnalysis"})
+@RequestMapping(path = "/work-types", produces = APPLICATION_JSON_VALUE)
+@RestController
+public class WorkTypesController extends BaseController {
+    private static final Logger LOG = getLogger(WorkTypesController.class);
+
+    private final TaskManagementService taskManagementService;
+    private final AccessControlService accessControlService;
+
+    public WorkTypesController(TaskManagementService taskManagementService,
+                               AccessControlService accessControlService) {
+        super();
+        this.taskManagementService = taskManagementService;
+        this.accessControlService = accessControlService;
+    }
+
+    @ApiOperation("Retrieve a list of work types for the current user")
+    @ApiResponses({
+        @ApiResponse(code = 200, message = OK, response = String.class)
+    })
+    @GetMapping(path = "/users")
+    public ResponseEntity<List<WorkType>> getWorkTypes(@RequestHeader(AUTHORIZATION) String authToken) {
+
+        List<WorkType> workTypes = new ArrayList<>();
+
+        AccessControlResponse roles = accessControlService.getRoles(authToken);
+
+        LOG.info("Access Control Response  found [{}]", roles);
+
+        if (roles.getRoleAssignments().isEmpty()) {
+            return ResponseEntity
+                .ok()
+                .cacheControl(CacheControl.noCache())
+                .body(Collections.emptyList());
+        } else {
+            Set<String> roleWorkTypes = getActorWorkTypes(authToken, roles);
+
+            if (!roleWorkTypes.isEmpty()) {
+                for (String workTypeId : roleWorkTypes) {
+                    Optional<WorkType> optionalWorkType = taskManagementService.getWorkType(workTypeId.trim());
+
+                    optionalWorkType.ifPresent(workTypes::add);
+                }
+            }
+
+            return ResponseEntity
+                .ok()
+                .cacheControl(CacheControl.noCache())
+                .body(workTypes);
+        }
+    }
+
+    private Set<String> getActorWorkTypes(String authToken, AccessControlResponse accessControlResponse) {
+        Set<String> roleWorkTypes = new HashSet<>();
+
+        String actorId = accessControlResponse.getRoleAssignments().get(0).getActorId();
+
+        AccessControlResponse actorRoles = accessControlService.getRolesByActorId(actorId, authToken);
+
```

Comment:
You've previously called:
accessControlService.getRoles(authToken);

which should do the same as this call do we need a secondary call here or can we obtain the date from the original accessControlResponse

Reply:
it looks AM team will only update the response for getByActor as mentioned in the following page:
https://tools.hmcts.net/confluence/pages/viewpage.action?pageId=1507735022&__ncforminfo=J86nJMaPNLB58DvRZfolWrdo1WFDBeGAw1nqVI2dI16SlSIY_W3Unxcb1-t9SoBvdvw8u0C_bhdZVo1Vbiu4WPRJ7HOTurI1#Userrolesandruleconfigurationrequirements-2.Worktypefilter
"Updated response of Get by Actor and Query Endpoint of RAS"